src/containers/CardContainer.tsx

```diff
@@ -0,0 +1,47 @@
+import React, {Component} from 'react';
+import Cardgrid from 'components/Cardgrid';
+
+export default class CardContainer extends Component<any, any> {
+
+  constructor(props: any) {
+    super(props);
+    this.state = {
+      data: [],
+      loading: true,
+    }
+    this.loadMore = this.loadMore.bind(this);
+  }
+
+  public componentDidMount() {
+    this.loadMore();
+  }
+
+  public loadMore() {
+    this.setState({loading: true});
+    const data = [];
+    const rand = Math.random();
+    for (let i = 0; i < 4; i++) {
+      data.push({
+        title: 'KTM 250SX 2017',
+        price: '$55$/d',
+        approx: '3 km away',
+        img: 'http://www.pngpix.com/wp-content/uploads/2016/07/PNGPIX-COM-Honda-CRF-450R-Motocross-Bike-PNG-Image.png',
+        key: rand + i,
+      })
+    }
+    setTimeout(() => this.setState({
+      data: this.state.data.concat(data),
+      loading: false,
+    }), 2000);
```

Comment:
I implemented this with `Promises` first, but then I decided to simplify it. It's just a fast mockup to show indicator.